src/proxy/inbound.rs

```diff
@@ -347,6 +342,18 @@ impl Inbound {
                     connection_security_policy: metrics::SecurityPolicy::mutual_tls,
                     destination_service: ds,
                 };
+
+                if !state.assert_rbac(&conn).await {
```

Comment:
the PR title (and lack of description) indicate this PR just adds metrics, but this seems like a change in behavior from before, since it's no longer in the else block

Reply:
if it's from a waypoint then why do we need to assert rbac now? before we did not, that seems correct

Reply:
That's a fair point, here's the issue: we need source info for metrics, and that is computed afterwards. I'm open to suggestions/alternatives here.